qiskit_nature/deprecation.py

```diff
@@ -228,6 +228,59 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def deprecate_positional_arguments(
+    version: str,
+    kw_pos_deprecated: Tuple[str],
+    additional_msg: Optional[str] = None,
+    stack_level: int = 3,
+) -> Callable:
+    """Decorator to convert positional arguments into keyword arguments and warn upon use.
+    .. code-block:: python
+
+        @deprecate_positional_arguments("0.1",(b,d))
+        def function(a,b,c,d,**kwargs):
+            returns a+c+sum(kwargs.values())
+
+        #This two calls would be equivalent
+        function(1,2,3,4) #Would raise deprecation warning for arguments b and d.
+        function(a=1,b=2,c=3,d=4) #Would not raise any warning.
+
+    Args:
+        version: Version to be used
+        kw_pos_deprecated: Tuple of arguments to be deprecated
+        additional_msg: any additional message
+        stack_level: stack level
+
+    Returns:
+        The decorated function
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            keyword_list = inspect.getfullargspec(func)[0]
+            for i, arg in enumerate(args):
+                kwargs[keyword_list[i]] = arg
+                if keyword_list[i] in kw_pos_deprecated:
+                    msg = (
+                        f"Please, refer from using '{keyword_list[i]}' positionally to"
+                        f"construct a VQEFactory. As of version {version} this are no longer"
+                        " properties of the factories but are passed to the VQE via keyword"
+                        " arguments. The option to pass the arguments positionally will be "
+                        "removed no sooner than 3 months after release."
```

Comment:
Please remove hard-coded mentioning of VQE and reword this to be more inline with the other messages raised by this module.

Reply:
Good catch

Reply:
By the way, what version should I indicate in the decorator?

Reply:
The message now looks like this:
```python
f"{func_name}: {keyword_list[i]} is no longer a positional argument "
f"as of version {version} and will be removed no sooner "
"than 3 months after the release. Instead use it as a keyword argument"
```
And we can add some additionla text if wanted.